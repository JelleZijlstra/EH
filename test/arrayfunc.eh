#!/usr/bin/ehi
include 'arrayfunc_lib.eh'

printvar arrayfunc([1, 2, 3, 4], func: n; ret 2 * n; end)
printvar arrayfunc(["foo", "bar", "baz"], func: str; ret str->0; end)
# Didn't even know that ehi actually accepts this
printvar arrayfunc([true, 1], (x => x.isA Bool))
# And even this
printvar arrayfunc([true, 1], (x => x.isA Bool))
