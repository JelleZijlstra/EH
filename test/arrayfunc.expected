%%stdout%%
@array [
0 => @int 2
1 => @int 4
2 => @int 6
3 => @int 8
]
@array [
0 => @string 'f'
1 => @string 'b'
2 => @string 'b'
]
@array [
0 => @bool true
1 => @bool false
]
@array [
0 => @bool true
1 => @bool false
]
%%stderr%%
